cardano-cli/src/Cardano/CLI/Json/Friendly.hs

```diff
@@ -172,7 +176,16 @@ friendlyTxBodyImpl
     , "update proposal" .= friendlyUpdateProposal txUpdateProposal
     , "validity range" .= friendlyValidityRange era (txValidityLowerBound, txValidityUpperBound)
     , "withdrawals" .= friendlyWithdrawals txWithdrawals
-    ]
+    ])
+
+redeemerIfShelleyBased :: CardanoEra era -> TxBody era -> [Aeson.Pair]
+redeemerIfShelleyBased era tb = caseByronOrShelleyBasedEra [] (\shEra -> [ "redeemers" .= friendlyRedeemer shEra tb ] ) era
+
+friendlyRedeemer :: ShelleyBasedEra era -> TxBody era -> Aeson.Value
+friendlyRedeemer _ (ShelleyTxBody _ _ _ TxBodyNoScriptData _ _) = Aeson.Null
+friendlyRedeemer _ (ShelleyTxBody _ _ _ (TxBodyScriptData _ _ r) _ _) = encodingToJSON $ L.toCBOR r
+  where encodingToJSON :: Encoding -> Aeson.Value
+        encodingToJSON e = either (Aeson.String . Text.pack) id $ fromFlatTerm (decodeValue True) $ toFlatTerm e
```

Comment:
I'm not sure about this failure here - instead of an error an user will see an error text in the redeemers field. That may be confusing.

Have you tried to autogenerate `ToJSON` for `Redeemers`? [It has `Generic`](https://cardano-ledger.cardano.intersectmbo.org/cardano-ledger-alonzo/Cardano-Ledger-Alonzo-TxWits.html#t:Redeemers) instance so maybe simply
```haskell
instance ToJSON Redeemers
```
could work here?

Reply:
Hm, after a lot of trying, that doesn't seem to work. I can use `genericToJSON` to make up for `Redeemers` not having `ToJSON` but then `RedeemersRaw` doesn't have it either, so I don't see any easy way to doing the conversion from `cardano-cli`.  What may work would be to go to `cardano-ledger` and add the `ToJSON` instances there.

Reply:
Other possible approaches (alternatives to adding `ToJSON` to `cardano-ledger`):
- Crash when `decodeValue` crashes
- Implement our own `decodeValue` (or fork it) and make it exhaustive
- Printing the result of showing the `Encoding` type
- Having a sum type in the JSON: something like `{decodedRedeemer = {...}}` and having `{error = "Could not decode because blah..."}`

Reply:
You're right, it's not easy to generate automatic `ToJSON` here. Thank you for checking that. I think your solution is the best for what I can think of. You can also ask on #ledger if they know if there's a better way to print `Redeemers`.

With regards to the failure reporting, I think crashing here would be too drastic. What do you think about printing the error to `stderr` and setting the field to null value?
(You'd need to change the type to return the possible error besides the JSON value).

Reply:
I see, because it is `stderr` it won't mess with `stdout`. I will give that a try

Reply:
Ok, done [in this PR](https://github.com/IntersectMBO/cardano-cli/pull/664/commits/dccd05e0cf80706ab7c942932712c68a16ba502f) 😎. I had a bit of fun, but I documented everything as well as I could.